### DISCUSSION TITLE
LootMode Questions-RAID Boss&Mobs drops no loot

### DISCUSSION THREAD

**Discussion by ArkknightX** [TrinityCore/TrinityCore#27254] [Q&A] [answered]
Hi everyone, 
I hope you are all well. I just joined Trinity Github after trying to do some fixings on my own 9.1.0 Shadowlands SQL Database.

Version : Client WoW 9.1.0 v39653 
9.1 v40000 Repack by Kyrian/Escapeland, downloaded from their Youtube link.
Issue:  Raid mobs & bosses drop no loot.
Affected Raids: Throne of Thunder (All 4 raids in this series)
NPCs in question: (all the trash mobs also drops no loot)

-  Jin'rokh the Breaker
- Stormbringer Draz'kil
- Horridon

I have done the following troubleshooting to see if anything has changed:

Changed all droprates(Chance value) to 100 in the following tables in  KyrianWorld DB:
`(UPDATE XXXXX_loot_template SET chance=100;)`

- creature_loot_template - World DB
- gameobject_loot_template - World DB
- item_loot_template  - World DB
- prospecting_loot_template - World DB
- reference_loot_template - World DB

After, completely shut down and restarted the PC, restarted the SQL server, bnet server and world server. Still no loot from any of the mobs in the above raid. Now, every time my character (created brand-new character), level up to 255 via .levelup 255 command, then join the Throne of Thunder series Raid, immediately after I arrive into the raid instance, I get a notification saying "You are not in this instance's group. You will be teleported to the nearest graveyard in 60 seconds." Either I can continue until i get kicked out, or enable `.gm on` that temporarily stops me being kicked out (the eviction message reappears when i turn gm mode off), or,  I exit the raid instance, and walk back in then the eviction message is permanently gone. Regardless of what I do, even after I exit and re-enter the raid instance, with no eviction message, all NPCs drops no loot.

I suspect this is a problem with assigning a unique Raid Group ID value to every character that joins an instance, and because this is violating the loot permissions set by `Loot mode` setting, there's no loot being dropped by any mob, or is being dropped, but just not visible to my character.

Is there a table where each raid is configured? a table that can edit permissions such as who can loot, who cannot see loot, who are to be kicked out, etc? Or is it the 'loot mode' value that sets the the flag/setting/value that prevents certain players with certain instance ID unable to loot?

Here's the monster `Horridon`boss for example:
Ingame `.lookup creature Horridon` results are:

- 368476 - [Horridon (4)]
- 68476 - [Horridon]
- 69204 - [Spawn of Horridon]
- 768476 - [Horridon (7)]

This SQL query 
`SELECT * FROM creature_loot_template WHERE entry=368476 OR entry=69204 OR entry=768476 OR entry=68476;
`shows me :

![Horridon creatureloot](https://user-images.githubusercontent.com/93835310/140618099-b777a5a1-c581-4a70-be3f-2d3d338f4ec5.png)
![Horridon creaturetemplate](https://user-images.githubusercontent.com/93835310/140618532-24d85f9c-92d5-4690-a990-fddc0950eeb4.png)

The orange part of LootMode is my edit ; I have changed them to 11 and 15 (which i saw from a properly working boss npc from a different raid) to test, but still does not drop loot; default for horridon were all 3 or 16.
What does the `LootID` value do in the `creature_template`?

For `Jin'rokh the Breaker`
`.lookup creature Jin'rokh`
14902 - [Jin'rokh the Breaker]
52728 - [Jin'rokh the Breaker]
69465 - [Jin'rokh the Breaker]
369465 - [Jin'rokh the Breaker(4)]
769465 - [Jin'rokh the Breaker(7)]

Jin'rokh the Breaker in the Raid:
SpawnID 450459
Entry: 69465
Faction : 14
Loot: 69465
AIName:   ScriptName: boss_jinrokh

I find the Trinity wikipedia does not contain enough information about this nor the loot mode. 
I saw some pages like  [MySQLStoredProcedures](https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130196/MySQLStoredProcedures) where it mentions calls like 
`sp_MakeLootable`
but this seems to be beyond what we casual contributors can access.

Basically i googled a lot about `loot mode`, but I found literally nothing on what each `loot mode` does (1,2,4,8, etc) or how it can be combined because I see some entries like 15 (1+2+4+8). Can someone please kindly enlighten me on this? 
I would like to contribute to the community by editing/creating a page explaining what each `loot mode` does in great detail so others who are facing same kind of technical issue like myself can refer, learn and fix the issue themselves.

Sorry for the wall post, I'd greatly appreciate some guidance here so I can also be more self-sufficient.
Let me know if you need any more details and I'll be happy to provide.
Thanks in advance everyone, much appreciated for what you're doing for rest of us.

**Comment by Aokromes** [comment, marked as answer]
we don't support repacks.

**Reply by ArkknightX** [reply]
That's fine, is it possible to get more info on what LootMode does? A table with each bitmask mode would be appreciated.
